ui/app/src/components/ItemMegaMenu/ItemMegaMenu.tsx

```diff
@@ -0,0 +1,276 @@
+            <div className={classes.itemState}>
+              <ItemPublishingTargetIcon item={item} className={classes.icon} />
+              <Typography variant="body2" component="span">
+                {getItemPublishingTargetText(item?.stateMap)}
+              </Typography>
+              <ItemStateIcon item={item} className={classes.icon} />
+              <Typography variant="body2" component="span">
+                {getItemStateText(item?.stateMap)}
+              </Typography>
+            </div>
+          </MenuItem>,
+          <div key={1} className={clsx(classes.mainItem, classes.actionsContainer)}>
+            <MenuList className={clsx(classes.actionsColumn, classes.itemsList)}>
+              {editorialOptions.map((option: MenuOption, y: number) => (
+                <MenuItem
+                  dense
+                  key={option.id}
+                  onClick={(e) => onMenuItemClicked(option.id, e)}
+                  className={clsx(classes.optionItem, propClasses?.menuItem)}
+                  children={option.label}
+                />
+              ))}
+            </MenuList>
+            <div className={classes.actionsColumn}>
+              {nonEditorialOptions.map((section: any, i: number) => (
+                <MenuList key={i} className={classes.itemsList}>
+                  {section.map((option: MenuOption, y: number) => (
+                    <MenuItem
+                      dense
+                      key={option.id}
+                      divider={i !== nonEditorialOptions.length - 1 && y === section.length - 1}
+                      onClick={(e) => onMenuItemClicked(option.id, e)}
+                      className={clsx(classes.optionItem, propClasses?.menuItem)}
+                      children={option.label}
+                    />
+                  ))}
+                </MenuList>
+              ))}
+            </div>
+          </div>,
+          <MenuItem key={2} className={clsx(classes.itemEdited, classes.infoItem, classes.mainItem)}>
+            <Typography variant="body2" component="h2">
+              <FormattedMessage
+                id="itemMegaMenu.editedBy"
+                defaultMessage="{edited} {date} {byLabel} {by}"
+                values={{
+                  date: new Intl.DateTimeFormat(locale.localeCode, locale.dateTimeFormatOptions).format(
+                    new Date(editedDate)
+                  ),
+                  by: editedBy,
+                  edited: (
+                    <span className={classes.itemEditedText}>
+                      <FormattedMessage id="words.edited" defaultMessage="Edited" />
+                    </span>
+                  ),
+                  byLabel: (
+                    <span className={classes.itemEditedText}>
+                      <FormattedMessage id="words.by" defaultMessage="by" />
+                    </span>
+                  )
+                }}
+              />
```

Comment:
Must handle the cases of unavailable pieces of this string (e.g. a null `by`, etc)